dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/file/batch/BatchFileHandler.kt

```diff
@@ -191,58 +189,104 @@ internal class BatchFileHandler(
         return file.renameToSafe(destFile)
     }
 
-    @Suppress("UnsafeThirdPartyFunctionCall") // Called within a try/catch block
-    private fun readEventHeader(stream: InputStream): Pair<EventMeta, Int>? {
-        val version = stream.read()
-        if (version < 0) {
-            internalLogger.e(ERROR_EOF_AT_VERSION_BYTE)
-            return null
+    @Throws(IOException::class)
+    private fun readBlock(stream: InputStream, expectedBlockType: BlockType): BlockReadResult {
```

Comment:
I decided to keep metadata, because there are some classes already created for it and didn't want to remove it (yes, we can get things from git history, but still).

Not sure I got your comment right: what is the benefit of postponing the validation?

Reply:
That would make reading more flexible, if you want to support stream like `[event][event][event][event]` and later `[meta][event][meta][event][meta][event]`. But also to support a stream with multiple metadata versions, which will happen when updating the SDK, `[meta1][event][meta2][event][meta2][event]`. It's best to read the block and do whatever you want based on the type you found.

Reply:
I'd say let's keep the metadata. We will definitely need it for the session replay persister for example.